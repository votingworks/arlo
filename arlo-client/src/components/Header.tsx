import React from 'react'
import styled from 'styled-components'
import {
  Navbar,
  NavbarGroup,
  NavbarHeading,
  Alignment,
} from '@blueprintjs/core'

const ButtonBar = styled.div`
  display: inline-block;
  margin-right: 10px;
`

const Nav = styled(Navbar)`
  width: 100%;

  .bp3-navbar-heading img {
    height: 35px;
    padding-top: 8px;
  }
`

const Header: React.FC<{}> = () => (
  <Nav fixedToTop>
    <NavbarGroup align={Alignment.LEFT}>
      <NavbarHeading>
        <img src="/arlo.png" alt="Arlo, by VotingWorks" />
      </NavbarHeading>
    </NavbarGroup>
    <NavbarGroup align={Alignment.RIGHT}>
      <ButtonBar id="reset-button-wrapper" />
    </NavbarGroup>
  </Nav>
)

export default Header
