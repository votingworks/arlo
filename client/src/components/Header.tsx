import React from 'react'
import styled from 'styled-components'
import {
  Navbar,
  NavbarGroup,
  NavbarHeading,
  Alignment,
  NavbarDivider,
} from '@blueprintjs/core'
import { Link, useRouteMatch, RouteComponentProps } from 'react-router-dom'
import { useAuthDataContext } from './UserContext'
import FormButton from './Atoms/Form/FormButton'
import { Inner } from './Atoms/Wrapper'

const ButtonBar = styled.div`
  display: inline-block;
  margin-right: 10px;
`

const Nav = styled(Navbar)`
  width: 100%;
  padding: 0;

  .bp3-navbar-heading img {
    height: 35px;
    padding-top: 5px;
  }

  .bp3-navbar-divider {
    margin-right: 15px;
    margin-left: 0;
  }
`

const InnerBar = styled(Inner)`
  display: inherit;
`

interface TParams {
  electionId: string
  jurisdictionId?: string
}

const Header: React.FC<{}> = () => {
  const electionMatch:
    | RouteComponentProps<TParams>['match']
    | null = useRouteMatch('/election/:electionId')
  const jurisdictionMatch:
    | RouteComponentProps<TParams>['match']
    | null = useRouteMatch(
    '/election/:electionId/jurisdiction/:jurisdictionId?'
  )
  const { isAuthenticated, meta } = useAuthDataContext()
  const electionId = electionMatch ? electionMatch.params.electionId : undefined
  const jurisdiction =
    jurisdictionMatch && meta
      ? meta.jurisdictions.find(
          j => j.id === jurisdictionMatch.params.jurisdictionId
        )
      : undefined
  return (
    <Nav>
      <InnerBar>
        <NavbarGroup align={Alignment.LEFT}>
          <NavbarHeading>
            <Link to="/">
              <img src="/arlo.png" alt="Arlo, by VotingWorks" />
            </Link>
          </NavbarHeading>
          {isAuthenticated && meta!.type === 'jurisdiction_admin' && (
            <NavbarHeading>
              {jurisdiction && `Jurisdiction: ${jurisdiction.name}`}
            </NavbarHeading>
          )}
        </NavbarGroup>
        <NavbarGroup align={Alignment.RIGHT}>
          {isAuthenticated && electionId && meta!.type === 'audit_admin' && (
            <>
              <NavbarHeading>
                <Link to={`/election/${electionId}/setup`}>Audit Setup</Link>
              </NavbarHeading>
              <NavbarDivider />
              <NavbarHeading>
                <Link to={`/election/${electionId}/progress`}>
                  Audit Progress
                </Link>
              </NavbarHeading>
              <NavbarDivider />
              <NavbarHeading>
                <Link to="/">View Audits</Link>
              </NavbarHeading>
              <NavbarDivider />
              <NavbarHeading>
                <Link to="/">New Audit</Link>
              </NavbarHeading>
            </>
          )}
          <ButtonBar id="reset-button-wrapper" />
          {/* istanbul ignore next */
          isAuthenticated && (
            <FormButton
              size="sm"
              onClick={
                /* istanbul ignore next */
                () => window.location.replace('/auth/logout')
              }
            >
              Log out
            </FormButton>
          )}
        </NavbarGroup>
      </InnerBar>
    </Nav>
  )
}

export default Header
