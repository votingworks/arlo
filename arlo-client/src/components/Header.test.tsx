import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'

it('renders correctly', () => {
  const { container } = render(<Header />)
  expect(container).toMatchSnapshot()
})
