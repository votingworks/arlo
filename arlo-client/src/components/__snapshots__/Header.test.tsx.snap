// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="bp3-navbar bp3-fixed-top sc-bwzfXH rpDk"
  >
    <div
      class="bp3-navbar-group bp3-align-left"
    >
      <div
        class="bp3-navbar-heading"
      >
        <img
          alt="Arlo, by VotingWorks"
          src="/arlo.png"
        />
      </div>
    </div>
    <div
      class="bp3-navbar-group bp3-align-right"
    >
      <div
        class="sc-bdVaJa bKktsd"
        id="reset-button-wrapper"
      />
    </div>
  </div>
</div>
`;
